cogs/gamerwords.py

```diff
@@ -52,23 +52,116 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
 # THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+import random
 import collections
 import re
+import asyncio
+from string import ascii_letters
+
 import discord
 from discord.ext import commands
-import asyncio
 import unidecode
-import random
+
 
 #GAMER_REGEX = r'(?:[^s]|^)(n+\s*i+\s*g\s*g+\s*(?:e+\s*r+|(a)))(\s*s+)?'
 GAMER_REGEX = r'(b+\s*r+\s*u+\s*h+)'
 
 with open('data/catchphrases.txt') as f:
 	CATCHPHRASES = list(map(str.rstrip, f))
 
+
+class UnicodeAwareBruhReplacer:
+	def __init__(self):
+		# match related
+		self.start_index = -1
+		self.letter_check = [False] * 4
+		self.match_length = 0
+		self.spaces = 0
+
+	def reset(self):
+		self.start_index = -1
+		self.letter_check = [False] * 4
+		self.match_length = 0
+		self.spaces = 0
+
+	def replace(self, text):
+		self.reset()
```

Comment:
I think a new replacer should be instantiated each time, given that there is no shared state that needs to persist per call to replace().

Reply:
My suggestion is therefore to remove this line. Possibly also to refuse running if it's been run before (like file objects, you could have a self.closed variable that is set to true at end and checked at beginning).